### PR TITLE
[ROCm][XLA] Adding three passes to  amdgpu compiler

### PR DESCRIPTION
**Pull request by jerryyin**
This CL (comes from discussions in #35991) introduced three passes to keep amdgpu_compiler in sync with nvptx_compiler:

- `reduction_degernerate_dim_remover`
- `reduction_dimension_grouper`
- `reduction_layout_normalizer`

@whchung @cheshire

### CODE REVIEW

**Review by cheshire — APPROVED** [review]

**Comment by jerryyin** [comment]
@cheshire Is it causing some internal failures? I don't have visibility in the `import/copybara` tab.

**Comment by cheshire** [comment]
That's strange, let me try again.

**Review by cheshire — APPROVED** [review]
